### DISCUSSION TITLE
How to catch exception or receive the result of transaction

### DISCUSSION THREAD

**Discussion by nhannh-senspark** [ton-org/game-engines-sdk#14] [Q&A] [answered]
Hi, I'm having some trouble and would love to get some help.

My question is after calling GameFi.buy() or buyWithTon(), how do I know what the status and result are so I can proceed? I tried Cancel transaction and got the following error:
```
Uncaught (in promise) Error: Wallet declined the request
at SendTransactionParser.parseAndThrowError (@tonconnect_ui-react.js?v=0d7dcad2:422:11)
at TonConnect.<anonymous> (@tonconnect_ui-react.js?v=0d7dcad2:2587:38)
```

Thank you.

**Comment by barinbritva** [comment, marked as answer]
@nhannh-senspark hi,

We can modify the `buy` method written at:
https://github.com/ton-community/flappy-bird/blob/ff2287d668e1d41c76c02201f80c7aa7f1181789/workspaces/client/src/ui.ts#L96

```typescript
try {
    await this.gameFi.buyWithJetton({
        amount: BigInt(price),
        forwardAmount: BigInt(1),
        forwardPayload: (window as any).Telegram.WebApp.initDataUnsafe.user.id + ':' + itemId
    });

    // Here we know transaction was sent. But it doesn't mean it received
    // Receiving mechanism is implemented on the backend side, in `processTxsForever` function
} catch (error) {
    console.error('Failed to buy an item.', error);
}
```

1. We added `await` to `buyWithJetton` call. In the next line we know transaction was sent
2. We added `catch` block to handle same errors
3. To handle some others errors or events you need to subscribe for them. Here is the doc: https://www.npmjs.com/package/@tonconnect/ui#tracking. I believe you need to subscribe for `transaction-signing-failed` event

**Reply by nhannh-senspark** [reply]
I really appreciate your feedback. Because of the urgent time for the project, I actually had another solution.
But you gave me new information that I really needed, it was very helpful.
Thank you very much, thank you for your project.
Best regards

**Reply by barinbritva** [reply]
Thanks a lot! I wish you all the best with your projects

**Comment by yjk596286200** [comment]
Hi，I met the same question. How did you resolve this problem. Could you give me some help.
Thank you very much,
